pkg/util/azure/azure.go

```diff
@@ -115,3 +119,55 @@ func getResponse(url string) (string, error) {
 
 	return string(all), nil
 }
+
+// GetHostname returns hostname based on Azure instance metadata.
+func GetHostname() (string, error) {
+	if !config.IsCloudProviderEnabled(CloudProviderName) {
+		return "", fmt.Errorf("cloud provider is disabled by configuration")
+	}
+
+	return getHostnameWithConfig(config.Datadog)
+}
+
+func getHostnameWithConfig(config config.Config) (string, error) {
+	style := config.GetString(hostnameStyleSetting)
+
+	if style == "os" {
+		return "", fmt.Errorf("azure_hostname_style is set to 'os'")
```

Comment:
Since the config default is `os` won't this error be triggered anytime this is not intentionally configured? It looks like this will only show up in expvars but was curious if this is desired.

Reply:
It also ends up in the debug log. I think it may be better to over-report these things, this way entire hostname decision process can be seen from the log alone (and consistent with how we always report unreachable GCE endpoint or disabled metadata provider). But I don't have a strong opinion here, if you think this might be an issue, this particular case can be silenced.